src/overlays/actors/ovl_Arrow_Ice/z_arrow_ice.c

```diff
@@ -28,33 +36,206 @@ const ActorInit Arrow_Ice_InitVars = {
     (ActorFunc)ArrowIce_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_809241F0[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_809241F0[];
-
-extern UNK_TYPE D_0E0002E0;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/D_80924200.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/ArrowIce_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/ArrowIce_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/ArrowIce_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/func_809224DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/func_809225D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/func_80922628.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/func_809227F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/ArrowIce_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Arrow_Ice/ArrowIce_Draw.s")
+extern Gfx D_0E0002E0[];
+
+void ArrowIce_SetupAction(ArrowIce* this, ArrowIceActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ArrowIce_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ArrowIce* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->radius = 0;
+    this->unk_158 = 1.0f;
+    ArrowIce_SetupAction(this, ArrowIce_Charge);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->alpha = 100;
+    this->timer = 0;
+    this->unk_15C = 0.0f;
+}
+
+void ArrowIce_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    func_80115D5C(&globalCtx->state);
+    (void)"消滅"; // Unreferenced in retail, means "Disappearance"
+}
+
+void ArrowIce_Charge(ArrowIce* this, GlobalContext* globalCtx) {
+    EnArrow* arrow = (EnArrow*)this->actor.parent;
+
+    if ((arrow == NULL) || (arrow->actor.update == NULL)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    if (this->radius < 10) {
+        this->radius++;
+    }
+    // copy position and rotation from arrow
+    this->actor.world.pos = arrow->actor.world.pos;
+    this->actor.shape.rot = arrow->actor.shape.rot;
+
+    func_800B9010(&this->actor, NA_SE_PL_ARROW_CHARGE_ICE - SFX_FLAG);
+
+    // if arrow has no parent, player has fired the arrow
+    if (arrow->actor.parent == NULL) {
+        this->unkPos = this->actor.world.pos;
+        this->radius = 10;
+        ArrowIce_SetupAction(this, ArrowIce_Fly);
+        this->alpha = 255;
+    }
+}
+
+void func_809225D0(Vec3f* unkPos, Vec3f* icePos, f32 scale) {
+    VEC3F_LERPIMPDST(unkPos, unkPos, icePos, scale);
+}
+
+void ArrowIce_Hit(ArrowIce* this, GlobalContext* globalCtx) {
+    f32 scale;
+    f32 offset;
+    u16 timer;
+
+    if (this->actor.projectedW < 50.0f) {
+        scale = 10.0f;
+    } else if (950.0f < this->actor.projectedW) {
```

Comment:
```suggestion
    } else if (this->actor.projectedW > 950.0f) {
```
My styling preference is to not have a literal at the start of a conditional unless it's necessary to match